src/cunumeric/scan/scan_local_util.h

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "cunumeric/cunumeric.h"
+
+#include <thrust/functional.h>
+
+namespace cunumeric {
+
+enum class ScanCode : int {
+  PROD = CUNUMERIC_SCAN_PROD,
+  SUM  = CUNUMERIC_SCAN_SUM,
+};
+
+template <typename Functor, typename... Fnargs>
+constexpr decltype(auto) op_dispatch(ScanCode op_code,
+                                     bool nan_to_identity,
+                                     Functor f,
+                                     Fnargs&&... args)
+{
+  switch (op_code) {
+    case ScanCode::PROD:
+      switch (nan_to_identity) {
```

Comment:
I guess you could simply use `if ... else ...` here.

Reply:
9d3e545ec7d6f137e937c0b48b1c256b8721a4e0